syntax = "proto3";

package grpc;

service Chat{
    rpc Connect(ChatUser) returns (ChatUserConnected);
    rpc Disconnect(ChatUser) returns (ChatUserConnected);
    rpc SendMessage(Message) returns (Empty);
    rpc SubscribeMessages(ChatUser) returns (stream Message);
}

message TimeStamp{
    int64 seconds = 1;
}

message Message{
    string senderID = 1;
    string message = 2;
    string conversationID = 3;
    TimeStamp timestamp = 4;
}

message Empty {}

message ChatUserConnected{
    bool isConnected = 1;
    string conversationID = 2;
}

message ChatUser{
    string username = 1;
    string conversationID = 2;
}